core/src/serve_repair.rs

```diff
@@ -338,17 +338,21 @@ pub(crate) struct RepairPeers {
 
 impl RepairPeers {
     fn new(asof: Instant, peers: &[ContactInfo], weights: &[u64]) -> Result<Self> {
-        if peers.is_empty() {
-            return Err(Error::from(ClusterInfoError::NoPeers));
-        }
         if peers.len() != weights.len() {
             return Err(Error::from(WeightedError::InvalidWeight));
         }
-        let weighted_index = WeightedIndex::new(weights)?;
-        let peers = peers
+        let (peers, weights): (Vec<_>, Vec<u64>) = peers
             .iter()
-            .map(|peer| (peer.id, peer.serve_repair))
-            .collect();
+            .zip(weights)
+            .filter_map(|(peer, &weight)| {
+                let addr = peer.serve_repair().ok()?;
+                Some(((peer.id, addr), weight))
+            })
+            .unzip();
```

Comment:
Given that we're not acting on the elements of weights, do we even need to zip/unzip it in with this iteration?

Reply:
`peers` and `weights` should be aligned; ie `peers[i]` corresponds to `weights[i]`;
The code is filtering out some of the peers; so it should filter out corresponding weights to keep them aligned.

Reply:
Ahh yep, you're right. My mind jumped over that `?` operator might return a `None`